File: 프로그래머스/lv1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    stack = []
    
    for a in arr:
        # 리스트 슬라이싱을 사용하면 비어있는 리스트에 대해서도 오류 없이 비교할 수 있음
        if stack[-1:] == [a]:
            continue
        stack.append(a)
    
    return stack

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        int[] answer = {};
        
        List<Integer> result = new ArrayList<>();
        result.add(arr[0]);
        
        for (int i = 1; i < arr.length; i++) {
            if (arr[i] != arr[i-1]) {
                result.add(arr[i]);
            }
        }
            
        // 리스트를 배열로 변환해서 반환
        return result.stream().mapToInt(i -> i).toArray();
    }
}